### ISSUE TITLE
Populate on virtual path with foreignField=_id doesn't work.

### ISSUE THREAD

**Issue by acrazing** [Automattic/mongoose#4684]
The test code as follow:

```ts
/*!
 *
 * Copyright 2016 - acrazing
 *
 * @author acrazing <joking.young@gmail.com>
 * @since 2016-11-03 14:51:07
 * @version 1.0.0
 * @desc populate.ts
 */


import { Schema, model, Document, SchemaTypes, disconnect } from 'mongoose';
import { connDb } from './base';
import { ObjectID } from 'mongodb';


interface Person extends Document {
  name?: string;
  bandId?: string;
  band_id?: string;
  band?: Band;
  band_?: string;
}


interface Band extends Document {
  name?: string;
  bid?: string;
  member?: Person;
}


const BandSchema = new Schema({
  name: String,
  bid : SchemaTypes.ObjectId,
}, { toJSON: { virtuals: true } });

const Band = model<Band>('Band', BandSchema);


const PersonSchema = new Schema({
  name   : String,
  bandId : SchemaTypes.ObjectId,
  band_id: SchemaTypes.ObjectId,
}, { toJSON: { virtuals: true } });

PersonSchema.virtual('band', {
  ref         : 'Band',
  localField  : 'bandId',
  foreignField: 'bid',
  justOne     : true,
})
PersonSchema.virtual('band_', {
  ref         : 'Band',
  localField  : 'band_id',
  foreignField: '_id',
  justOne     : true,
})
const Person = model<Person>('Person', PersonSchema);

connDb().then(() => Promise.all([
  Person.remove({}).exec(),
  Band.remove({}).exec(),
])).then(() => new Band({
  name: 'bandA',
  bid : new ObjectID,
}).save()).then((band) => {
  console.log('band', band.bid, band._id)
  return new Person({
    name   : 'personA',
    bandId : band.bid,
    band_id: band._id,
  }).save()
}).then(() => Person.find({}).populate('band band_').exec()).then((data) => {
  console.log('person.band', data[0].toJSON())
  return data[0]
}).then((person) => Person.find({ _id: { $in: [person._id] } }).exec()).then((data) => {
  console.log('person.find._id.$in', data)
}).then(() => disconnect())
```

execute output:

```bash
+ ts-node ./src/__test__/populate.ts
band 581d458365de7c7c0ab595fd 581d458365de7c7c0ab595fe
person.band { _id: 581d458365de7c7c0ab595ff,
  name: 'personA',
  bandId: 581d458365de7c7c0ab595fd,
  band_id: 581d458365de7c7c0ab595fe,
  __v: 0,
  band_: undefined, # assert this is same to `band` field
  band: 
   { _id: 581d458365de7c7c0ab595fe,
     name: 'bandA',
     bid: 581d458365de7c7c0ab595fd,
     __v: 0,
     id: '581d458365de7c7c0ab595fe' },
  id: '581d458365de7c7c0ab595ff' }
person.find._id.$in [ { _id: 581d458365de7c7c0ab595ff,
    name: 'personA',
    bandId: 581d458365de7c7c0ab595fd,
    band_id: 581d458365de7c7c0ab595fe,
    __v: 0 } ]
```

env:

```bash
+ node --version
v5.6.0
+ node -p -e 'require("mongoose/package.json").version'
4.6.5
```

**Comment by Jeff-Lewis** [comment]
Is there a typo in Person.populate('band band_')? Should it be band_id?

**Comment by acrazing** [comment]
Well, I found this is fixed in `4.6.6`.
